provider/filestore.go

```diff
@@ -581,23 +588,23 @@ func (fs *HDFSFileStore) ServeDirectory(files []filestore.Filepath) (Iterator, e
 
 func (fs *HDFSFileStore) Serve(files []filestore.Filepath) (Iterator, error) {
 	if len(files) == 0 {
-		return nil, fmt.Errorf("no files to serve")
+		return nil, fferr.NewInvalidArgument(fmt.Errorf("no files to serve"))
 	}
 	if len(files) > 1 {
 		return fs.ServeDirectory(files)
 	}
 	file := files[0]
 	b, err := fs.Client.ReadFile(file.Key())
 	if err != nil {
-		return nil, fmt.Errorf("could not read file: %w", err)
+		return nil, err
 	}
 	switch file.Ext() {
 	case filestore.Parquet:
 		return parquetIteratorFromBytes(b)
 	case filestore.CSV:
-		return nil, fmt.Errorf("could not find CSV reader")
+		return nil, fferr.NewInternalError(fmt.Errorf("CSV reader not implemented"))
 	default:
-		return nil, fmt.Errorf("unsupported file type")
+		return nil, fferr.NewDataTypeNotFoundError(string(file.Ext()), fmt.Errorf("unsupported file type"))
```

Comment:
This should be something different since its not a datatype error

Reply:
Invalid arg or internal?

Reply:
On the one hand, this could be an invalid arg if a user supplies a file and then directly tries to server it; otherwise, it seems like an internal error if Spark generates something other than parquet.

Reply:
Can this be both something that we set and the user set? We may want an InvalidFileFormat error